pkgs/tools/misc/shim/default.nix

```diff
@@ -0,0 +1,57 @@
+{ stdenv, fetchFromGitHub, lib, elfutils, vendorCertFile ? null
+, defaultLoader ? null, ... }:
+
+let
+
+  inherit (stdenv.targetPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
+  target = {
+    x86_64-linux = "shimx64.efi";
+    aarch64-linux = "shimaa64.efi";
+  }.${system} or throwSystem;
+in stdenv.mkDerivation rec {
+  pname = "shim";
+  version = "15.7";
+
+  src = fetchFromGitHub {
+    owner = "rhboot";
+    repo = pname;
+    rev = version;
+    hash = "sha256-TRSpyZC4pt1wU3pxQ0WV6ARpPWPd3qwiJCtilzSLLXQ=";
```

Comment:
You could just fetch the `gnu_efi_src` submodule here, instead of going through the pain of patching it into the source manually. Fetching it separately only makes sense when it can also build as a separate derivation.